File: 贪心算法/12.py
class Solution:
    def intToRoman(self, num: int) -> str:
        dic = {1000:"M",900:"CM",500:"D",400:"CD",100:"C",90:"XC",50:"L",40:"XL",10:"X",9:"IX",5:"V",4:"IV",1:"I"}
        res = []
        for key in dic.keys():
            while(num//key>0):
                res.append(dic[key])
                num -= key
        return "".join(res)

num = 2984
obj = Solution()
print(obj.intToRoman(num))

File: 贪心算法/12.cpp
#include<string>
#include<vector>
using namespace std;
class Solution {
public:
    string intToRoman(int num) {
        vector<int> nums = {1000,900,500,400,100,90,50,40,5,4,1};
        vector<string> roman = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        string res;
        for(int i=0;i<13;i++)
            while(num/nums[i]>0)
            {
                num -= nums[i];
                res += roman[i];
            }
        return res;
    }
};